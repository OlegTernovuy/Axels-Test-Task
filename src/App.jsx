import "./App.css";
import { Footer, Header } from "./components";
import ProductsPage from "./pages/productsPage/ProductsPage";

function App() {
  return (
    <div className="App">
      <Header/>
      <ProductsPage />
      <Footer/>
    </div>
  );
}

export default App;
