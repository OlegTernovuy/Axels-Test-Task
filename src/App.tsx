import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import ProductsPage from './pages/ProductsPage';
import NotFound from './pages/NotFound';
import { Footer, Header, ProductModal } from './components';
import { AppDiv } from './styled/StyledApp';

const routes = createBrowserRouter([
    {
        path: '/',
        element: <ProductsPage />,
        errorElement: <NotFound />,
        children: [
            {
                path: 'products/:id',
                element: <ProductModal />,
            },
        ],
    },
]);

function App() {
    return (
        <AppDiv>
            <Header />
            <RouterProvider router={routes} />
            <Footer />
        </AppDiv>
    );
}

export default App;
