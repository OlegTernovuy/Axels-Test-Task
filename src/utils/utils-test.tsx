import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { rootReducer } from '../redux';

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = configureStore({
            reducer: rootReducer,
            preloadedState,
        }),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }

    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
