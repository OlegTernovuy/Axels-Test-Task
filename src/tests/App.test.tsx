import App from '../App';

import { renderWithProviders } from './utils/utils-test';

describe('App component', () => {
    it('should match a snapshot', () => {
        const view = renderWithProviders(<App />);
        expect(view).toMatchSnapshot();
    });
});
