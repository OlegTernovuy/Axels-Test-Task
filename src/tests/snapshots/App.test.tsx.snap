// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jlZhRR dwJfrq"
      >
        <header
          class="sc-eqUzNf llCMqC"
        >
          <h1>
            Axels Test Task
          </h1>
          <span>
            Oleh Ternovyi
          </span>
        </header>
        <div
          class="sc-aYaIB ewgwFf"
        >
          <div
            class="MuiBox-root css-0"
          >
            <span>
              Loading...
            </span>
          </div>
        </div>
        <footer
          class="sc-fqkwJk eedRsB"
        >
          <span>
            Oleh Ternovyi 2024
          </span>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jlZhRR dwJfrq"
    >
      <header
        class="sc-eqUzNf llCMqC"
      >
        <h1>
          Axels Test Task
        </h1>
        <span>
          Oleh Ternovyi
        </span>
      </header>
      <div
        class="sc-aYaIB ewgwFf"
      >
        <div
          class="MuiBox-root css-0"
        >
          <span>
            Loading...
          </span>
        </div>
      </div>
      <footer
        class="sc-fqkwJk eedRsB"
      >
        <span>
          Oleh Ternovyi 2024
        </span>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
